pkg/infra/pulumi_aws/deploylib.ts

```diff
@@ -341,12 +341,15 @@ export class CloudCCLib {
         }
     }
 
-    createDockerAppRunner(execUnitName, envVars: any) {
+    createImage(execUnitName: string, dockerfilePath: string) {
         const image = this.sharedRepo.buildAndPushImage({
-            context: `./${execUnitName}`,
+            context: `./${execUnitName}/${dockerfilePath}`,
```

Comment:
I think this needs to be
```js
{
    context: `./${execUnitName}/${path.dirname(dockerfilePath)}`,
    dockerfile: `./${execUnitName}/${dockerfilePath}`,
}
```
Though I'm not sure if using that context is correct versus using `./${execUnitName}`.

Reply:
so i printed out what my code has for the context and its `    ./main/src` which would be correct

Reply:
we can leave context at the root of the exec unit if thats what you mean

Reply:
the path is already filepath.Dir in go so it doesnt include the file name

Reply:
Hm, isn't that wrong then? What if your dockerfile was `build/myservice.Dockerfile`? At the least the field name doesn't match its content.

Reply:
i might be a bit confused by what you mean. Doesnt it assume that there is 1 and only 1 dockerfile within the context and it will use that? if it was `build/myservice.Dockerfile` then we would point to `{execunit}/build/myservice.Dockerfile` which i would think is correct

Reply:
The `file` parameter defaults to `Dockerfile`, but I don't believe it's relative to the context but rather the current working directory from when it's run. I don't believe it does any sort of searching. Additionally, by sending the context of `{execunit}/build` that would mean it couldn't include any source files in `{execunit}/src` (or similar, outside of the build directory).

Reply:
theres no way we can understand the context without parsing the Dockerfile, so are you suggesting just leaving it at the root exec unit so we can encompass everything? If so then thats fine with me and similarly is the question, does the --file parameter only look for files named Dockerfile or would it find any Dockerfile?

Reply:
> theres no way we can understand the context without parsing the Dockerfile

Not only that, but even that might not be able to tell you enough (though in many circumstances we could make a guess based on any `ADD`/`COPY` statements). This would have to be a directive to support all the different edge cases.

> are you suggesting just leaving it at the root exec unit so we can encompass everything?

Yeah, for now until we get some more intelligence built in.

> does the --file parameter only look for files named Dockerfile or would it find any Dockerfile?

I'm relatively certain it only finds the file named `Dockerfile`. Most non-trivial build processes I've seen use explicit `-f path/to/dockerfile` for this reason.

Reply:
ok i can go make these updates then, will have a new revision out today